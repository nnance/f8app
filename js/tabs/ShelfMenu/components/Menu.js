import React from 'react';
import {
  View,
  Image,
  Text,
  TouchableOpacity,
  ScrollView,
} from 'react-native';

import ProfilePicture from '../../../common/ProfilePicture';
import { STATUS_BAR_HEIGHT, colors } from '../../../common/styles';
import { getCategoryIcon } from '../../../models/clog';

import ClogMenuItem from './ClogMenuItem';
import WhiteMenuItem from './WhiteMenuItem';
import FadedMenuItem from './FadedMenuItem';
import styles from './styles';

const CloseMenuButton = ({ onPress }) => (
  <TouchableOpacity onPress={onPress}>
    <View style={{ width: 40, height: 40, flex: 0, paddingLeft: 10, justifyContent: 'center', alignItems: 'flex-start' }}>
      <Image source={require('../../../assets/shelf-menu/close.png')} style={{ width: 18, height: 18, resizeMode: 'contain' }} />
    </View>
  </TouchableOpacity >
);

const Topic = ({ title, children, style }) => (
  <View>
    <View
      style={{
        backgroundColor: colors.greyBackground,
        paddingVertical: 4,
        paddingLeft: 10,
      }}
    >
      <Text style={{ color: 'rgba(255, 255, 255, 0.2)', fontSize: 12 }}>{title}</Text>
    </View>
    <View style={[{ paddingVertical: 0, paddingLeft: 10 }, style]}>
      {children}
    </View>
  </View>
);

class ShelfMenu extends React.Component {
  render() {
    const mockTags = [{ id: '1', title: 'Chat Clog' }, { id: '2', title: 'Video Clog' }, { id: '3', title: 'Picture Clog' }, { id: '4', title: 'Pet Clog' }];
    return (
      <View style={{ flex: 1, backgroundColor: 'rgba(50, 31, 85, 1)' }}>
        <View style={{ height: STATUS_BAR_HEIGHT }} />
        <View style={styles.closeContainer}>
          <CloseMenuButton onPress={this.props.onClose} />
        </View>
        <ScrollView>
          <Topic title="หมวดหมู่เรื่องราว">
            <ClogMenuItem
              onClogPress={this.props.onGagClogPress}
              tags={mockTags}
              title="Gag Clog"
              source={getCategoryIcon('G')}
            />
            <ClogMenuItem
              onClogPress={this.props.onDiaryClogPress}
              tags={mockTags}
              title="Diary Clog"
              source={getCategoryIcon('D')}
            />
            <ClogMenuItem
              onClogPress={this.props.onNovelClogPress}
              tags={mockTags}
              title="Novel Clog"
              source={getCategoryIcon('N')}
            />
            <ClogMenuItem
              notShowBottomLine
              onClogPress={this.props.onMythClogPress}
              tags={mockTags}
              title="Myth Clog"
              source={getCategoryIcon('M')}
            />
          </Topic>
          <Topic title="ข้อมูล">
            <WhiteMenuItem
              onPress={this.props.onProfilePress}
              title={this.props.user.name}
              icon={<ProfilePicture size={25} user={this.props.user} />}
            />
          </Topic>
          <Topic title="เกี่ยวกับ" style={{ paddingVertical: 5 }}>
            <FadedMenuItem onPress={this.props.onAboutClogiiPress} title="เกี่ยวกับ Clogii" source={require('../../../assets/shelf-menu/about.png')} />
            <FadedMenuItem onPress={this.props.onPolicyPress} title="ข้อกำหนดในการใช้ Clogii" source={require('../../../assets/shelf-menu/policy.png')} />
          </Topic>
          <Topic title="ติดตาม" style={{ paddingVertical: 5 }}>
            <FadedMenuItem onPress={this.props.onFacebookPress} title="Facebook" source={require('../../../assets/shelf-menu/fb.png')} />
            <FadedMenuItem onPress={this.props.onLinePress} title="Line" source={require('../../../assets/shelf-menu/line.png')} />
            <FadedMenuItem onPress={this.props.onInstagramPress} title="Instagram" source={require('../../../assets/shelf-menu/ig.png')} />
          </Topic>
        </ScrollView>
      </View>
    );
  }
}

export default ShelfMenu;
