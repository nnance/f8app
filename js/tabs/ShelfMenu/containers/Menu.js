import { connect } from 'react-redux';

import Menu from '../components/Menu';

const mapStateToProps = state => ({
  user: state.user,
});

export default connect(mapStateToProps)(Menu);
