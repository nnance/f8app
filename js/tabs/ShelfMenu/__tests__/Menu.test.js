import React from 'react';
import renderer from 'react-test-renderer';

import Menu from '../components/Menu';

const user = {
  name: 'ima',
};

describe('Menu', () => {
  it('render', () => {
    const tree = renderer.create(<Menu user={user} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
