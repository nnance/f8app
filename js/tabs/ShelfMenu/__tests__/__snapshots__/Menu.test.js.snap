exports[`Menu render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(50, 31, 85, 1)",
      "flex": 1,
    }
  }>
  <View
    style={
      Object {
        "height": 15,
      }
    } />
  <View
    style={
      Object {
        "paddingLeft": 15,
        "paddingVertical": 10,
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Image
        source={1}
        style={
          Object {
            "height": 15,
            "resizeMode": "contain",
            "width": 15,
          }
        } />
    </View>
  </View>
  <ScrollView>
    <View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.2)",
            "paddingLeft": 10,
            "paddingVertical": 4,
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(255, 255, 255, 0.2)",
              "fontSize": 12,
            }
          }>
          หมวดหมู่เรื่องราว
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "paddingLeft": 10,
              "paddingVertical": 0,
            },
            undefined,
          ]
        }>
        <View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "paddingLeft": 5,
                  "paddingVertical": 10,
                },
                Object {
                  "paddingBottom": 10,
                  "paddingTop": 10,
                },
              ]
            }>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "opacity": 1,
                }
              }
              testID={undefined}>
              <View
                style={
                  Object {
                    "height": 25,
                    "width": 25,
                  }
                }>
                <Image
                  source={1}
                  style={
                    Object {
                      "height": 25,
                      "resizeMode": "contain",
                      "width": 25,
                    }
                  } />
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                    "paddingLeft": 10,
                  }
                }>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "white",
                      "fontWeight": "bold",
                    }
                  }>
                  Gag Clog
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "alignItems": "flex-end",
                  "justifyContent": "center",
                  "marginRight": 10,
                  "width": 50,
                }
              }>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID={undefined}>
                <Image
                  source={1}
                  style={
                    Object {
                      "height": 20,
                      "resizeMode": "contain",
                      "width": 20,
                    }
                  } />
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.2)",
                "flex": 1,
                "height": 1,
                "marginLeft": 40,
                "marginVertical": 0,
              }
            } />
        </View>
        <View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "paddingLeft": 5,
                  "paddingVertical": 10,
                },
                Object {
                  "paddingBottom": 10,
                  "paddingTop": 10,
                },
              ]
            }>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "opacity": 1,
                }
              }
              testID={undefined}>
              <View
                style={
                  Object {
                    "height": 25,
                    "width": 25,
                  }
                }>
                <Image
                  source={1}
                  style={
                    Object {
                      "height": 25,
                      "resizeMode": "contain",
                      "width": 25,
                    }
                  } />
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                    "paddingLeft": 10,
                  }
                }>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "white",
                      "fontWeight": "bold",
                    }
                  }>
                  Diary Clog
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "alignItems": "flex-end",
                  "justifyContent": "center",
                  "marginRight": 10,
                  "width": 50,
                }
              }>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID={undefined}>
                <Image
                  source={1}
                  style={
                    Object {
                      "height": 20,
                      "resizeMode": "contain",
                      "width": 20,
                    }
                  } />
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.2)",
                "flex": 1,
                "height": 1,
                "marginLeft": 40,
                "marginVertical": 0,
              }
            } />
        </View>
        <View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "paddingLeft": 5,
                  "paddingVertical": 10,
                },
                Object {
                  "paddingBottom": 10,
                  "paddingTop": 10,
                },
              ]
            }>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "opacity": 1,
                }
              }
              testID={undefined}>
              <View
                style={
                  Object {
                    "height": 25,
                    "width": 25,
                  }
                }>
                <Image
                  source={1}
                  style={
                    Object {
                      "height": 25,
                      "resizeMode": "contain",
                      "width": 25,
                    }
                  } />
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                    "paddingLeft": 10,
                  }
                }>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "white",
                      "fontWeight": "bold",
                    }
                  }>
                  Novel Clog
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "alignItems": "flex-end",
                  "justifyContent": "center",
                  "marginRight": 10,
                  "width": 50,
                }
              }>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID={undefined}>
                <Image
                  source={1}
                  style={
                    Object {
                      "height": 20,
                      "resizeMode": "contain",
                      "width": 20,
                    }
                  } />
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.2)",
                "flex": 1,
                "height": 1,
                "marginLeft": 40,
                "marginVertical": 0,
              }
            } />
        </View>
        <View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "paddingLeft": 5,
                  "paddingVertical": 10,
                },
                Object {
                  "paddingBottom": 10,
                  "paddingTop": 10,
                },
              ]
            }>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "opacity": 1,
                }
              }
              testID={undefined}>
              <View
                style={
                  Object {
                    "height": 25,
                    "width": 25,
                  }
                }>
                <Image
                  source={1}
                  style={
                    Object {
                      "height": 25,
                      "resizeMode": "contain",
                      "width": 25,
                    }
                  } />
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                    "paddingLeft": 10,
                  }
                }>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "white",
                      "fontWeight": "bold",
                    }
                  }>
                  Myth Clog
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "alignItems": "flex-end",
                  "justifyContent": "center",
                  "marginRight": 10,
                  "width": 50,
                }
              }>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID={undefined}>
                <Image
                  source={1}
                  style={
                    Object {
                      "height": 20,
                      "resizeMode": "contain",
                      "width": 20,
                    }
                  } />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.2)",
            "paddingLeft": 10,
            "paddingVertical": 4,
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(255, 255, 255, 0.2)",
              "fontSize": 12,
            }
          }>
          ข้อมูล
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "paddingLeft": 10,
              "paddingVertical": 0,
            },
            undefined,
          ]
        }>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "paddingLeft": 5,
                "paddingVertical": 10,
              },
              undefined,
            ]
          }>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "opacity": 1,
              }
            }
            testID={undefined}>
            <View
              style={
                Object {
                  "height": 25,
                  "width": 25,
                }
              }>
              <View
                style={
                  Object {
                    "backgroundColor": "black",
                    "borderRadius": 12.5,
                    "height": 25,
                    "width": 25,
                  }
                } />
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "paddingLeft": 10,
                }
              }>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Object {
                    "color": "white",
                    "fontWeight": "bold",
                  }
                }>
                ima
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "justifyContent": "center",
                "marginRight": 10,
                "width": 50,
              }
            } />
        </View>
      </View>
    </View>
    <View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.2)",
            "paddingLeft": 10,
            "paddingVertical": 4,
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(255, 255, 255, 0.2)",
              "fontSize": 12,
            }
          }>
          เกี่ยวกับ
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "paddingLeft": 10,
              "paddingVertical": 0,
            },
            Object {
              "paddingVertical": 5,
            },
          ]
        }>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "opacity": 1,
              "paddingHorizontal": 5,
              "paddingVertical": 6,
            }
          }
          testID={undefined}>
          <Image
            source={1}
            style={
              Object {
                "height": 20,
                "resizeMode": "contain",
                "width": 20,
              }
            } />
          <View
            style={
              Object {
                "marginLeft": 10,
              }
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontSize": 13,
                  },
                  Object {
                    "color": "rgba(255, 255, 255, 0.2)",
                  },
                ]
              }>
              เกี่ยวกับ Clogii
            </Text>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "opacity": 1,
              "paddingHorizontal": 5,
              "paddingVertical": 6,
            }
          }
          testID={undefined}>
          <Image
            source={1}
            style={
              Object {
                "height": 20,
                "resizeMode": "contain",
                "width": 20,
              }
            } />
          <View
            style={
              Object {
                "marginLeft": 10,
              }
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontSize": 13,
                  },
                  Object {
                    "color": "rgba(255, 255, 255, 0.2)",
                  },
                ]
              }>
              ข้อกำหนดในการใช้ Clogii
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.2)",
            "paddingLeft": 10,
            "paddingVertical": 4,
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(255, 255, 255, 0.2)",
              "fontSize": 12,
            }
          }>
          ติดตาม
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "paddingLeft": 10,
              "paddingVertical": 0,
            },
            Object {
              "paddingVertical": 5,
            },
          ]
        }>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "opacity": 1,
              "paddingHorizontal": 5,
              "paddingVertical": 6,
            }
          }
          testID={undefined}>
          <Image
            source={1}
            style={
              Object {
                "height": 20,
                "resizeMode": "contain",
                "width": 20,
              }
            } />
          <View
            style={
              Object {
                "marginLeft": 10,
              }
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontSize": 13,
                  },
                  Object {
                    "color": "rgba(255, 255, 255, 0.2)",
                  },
                ]
              }>
              Facebook
            </Text>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "opacity": 1,
              "paddingHorizontal": 5,
              "paddingVertical": 6,
            }
          }
          testID={undefined}>
          <Image
            source={1}
            style={
              Object {
                "height": 20,
                "resizeMode": "contain",
                "width": 20,
              }
            } />
          <View
            style={
              Object {
                "marginLeft": 10,
              }
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontSize": 13,
                  },
                  Object {
                    "color": "rgba(255, 255, 255, 0.2)",
                  },
                ]
              }>
              Line
            </Text>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "opacity": 1,
              "paddingHorizontal": 5,
              "paddingVertical": 6,
            }
          }
          testID={undefined}>
          <Image
            source={1}
            style={
              Object {
                "height": 20,
                "resizeMode": "contain",
                "width": 20,
              }
            } />
          <View
            style={
              Object {
                "marginLeft": 10,
              }
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontSize": 13,
                  },
                  Object {
                    "color": "rgba(255, 255, 255, 0.2)",
                  },
                ]
              }>
              Instagram
            </Text>
          </View>
        </View>
      </View>
    </View>
  </ScrollView>
</View>
`;
